universal-application-tool-0.0.1/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java

```diff
@@ -275,17 +282,18 @@ private boolean showBlock(Block block) {
   }
 
   /** Returns the identifier of uploaded file if applicable. */
-  private Optional<String> getFileKey(ApplicantQuestion question) {
-    switch (question.getType()) {
-      case FILEUPLOAD:
-        FileUploadQuestion fileUploadQuestion = question.createFileUploadQuestion();
-        if (!fileUploadQuestion.isAnswered()) {
-          return Optional.empty();
-        }
-        return fileUploadQuestion.getFileKeyValue();
-      default:
-        return Optional.empty();
+  private Optional<String> getFileKey(Optional<FileUploadQuestion> question) {
+    if (question.isEmpty() || !question.get().isAnswered()) {
```

Comment:
Optional: you can make this method body the following.


return question.filter(FileUploadQuestion::isAnswered)
    .map(FileUploadQuestion::getFileKeyValue)
    .orElseGet(() -> Optional.empty());

Reply:
I learned through review like this :D but this is effectively the "Streams" feature added in Java8.  

Collections can all become streams with .stream(), and Optional added some of the filter/map concepts too.

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html